lib/parameter_substitution.rb

```diff
@@ -72,6 +72,10 @@ def find_formatters(string_with_tokens, mapping: {})
       parse_expression(context_from_string(string_with_tokens, mapping)).method_names
     end
 
+    def find_warnings(string_with_tokens, mapping: {}, warning_type: nil)
```

Comment:
It doesn't look like there are any callers of `find_warnings` that pass in a `warning_type` - is this param and special handling for different warning types needed?

Reply:
At one point I was calling this new method in `web` with that parameter for a certain scenario but one of the last things I did was remove that use when I fixed the json sub param quotes issue so I think that can be removed now. I don't see a scenario anymore where it's needed.

Reply:
Alright, `warning_type` param and logic has been removed and pushed up.

Reply:
Thanks! That looks cleaner. Can you also remove `UNKNOWN_PARAM_WARNING_TYPE` and `UNKNOWN_METHOD_WARNING_TYPE`?

Reply:
Removed and pushed up.

Reply:
👍